File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [0] * n
    q = deque()
    def bfs(start):
        q.append(start)
        while q:
            v = q.popleft()
            if visited[v] == 0:
                visited[v] = 1
            for i in range(len(computers)):
                if computers[v][i] == 1 and visited[i] == 0:
                    q.append(i)
    while 0 in visited:
        answer += 1
        bfs(visited.index(0))
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited = new Array(n).fill(0);
    const q = [];
    
    for (let i = 0; i < n; i++){
        if (visited[i]) continue;
        
        answer++;
        visited[i] = 1;
        q.push(i);
        
        while (q.length > 0) {
            const v = q.shift();
            for (let idx = 0; idx < n; idx++) {
                if (computers[v][idx] && !visited[idx]){
                    visited[idx] = 1
                    q.push(idx)    
                }
            }
        }
    }
    return answer;
}